src/hooks/useOptionsSupply.ts

```diff
@@ -0,0 +1,41 @@
+import { ethers } from "ethers";
+import { useQuery } from "react-query";
+
+import { useConnection } from "./useConnection";
+import { KPIOptionsToken, infuraId } from "../config";
+
+export function useOptionsSupply() {
+  const { provider, network } = useConnection();
+  const { data, isLoading, error } = useQuery<any>(
+    ["kpi options supply", provider, network?.chainId],
+    () => getOptionsSupply(provider, network)
+  );
+
+  const format = Intl.NumberFormat("en-US", {
+    maximumSignificantDigits: 1,
+  }).format;
+
+  return {
+    supply: data ? format(+ethers.utils.formatEther(data)) : undefined,
+    isLoading,
+    error,
+  };
+}
+
+async function getOptionsSupply(
+  web3Provider: ethers.providers.Web3Provider | null,
+  network: ethers.providers.Network | null
+) {
+  const chainId = (network && network.chainId ? network.chainId : 1) as 1 | 42;
+  const provider =
```

Comment:
`provider` here might be null (the user has not unlocked their wallet yet) but we still want to fetch the tokens supply with a fallback

Reply:
i see, that make sense. IMO if you are doing this, would it make more sense to create a default provider globally on page load, and then override with the users signer once available? maybe this would cause problems with a default infura provider though. this might make sense as a general pattern for dapps. WDYT? also im not suggesting to change to this, just curious

Reply:
i guess you also dont have chain id to do this